touchforms/backend/xformplayer.py

```diff
@@ -209,17 +209,12 @@ def walk(self):
 
     def _walk(self, parent_ix, siblings):
         def step(ix, descend):
-            next_ix = self.fem.incrementIndex(ix, descend)
+            next_ix = self.fec.getAdjacentIndex(ix, True, descend)
             self.fem.setQuestionIndex(next_ix)  # needed to trigger events in form engine
             return next_ix
 
         def ix_in_scope(form_ix):
-            if form_ix.isEndOfFormIndex():
-                return False
-            elif parent_ix.isBeginningOfFormIndex():
-                return True
-            else:
-                return FormIndex.isSubElement(parent_ix, form_ix)
+            return form_ix.isInForm()
```

Comment:
I definitely don't think this is correct since this method gets called recursively for groups and repeats so that `ix_in_scope` isn't just checking if the `form_ix` is in the form but if its in the scope of the current `parent_ix`.